database/query/datasources.sql

```diff
@@ -0,0 +1,73 @@
+-- CreateDataSource creates a new datasource in a given project.
+
+-- name: CreateDataSource :one
+INSERT INTO data_sources (project_id, name, display_name)
+VALUES ($1, $2, $3) RETURNING *;
+
+-- AddDataSourceFunction adds a function to a datasource.
+
+-- name: AddDataSourceFunction :one
+INSERT INTO data_sources_functions (data_source_id, name, type, definition)
+VALUES ($1, $2, $3, $4) RETURNING *;
```

Comment:
**issue:** `INSERTS` should always specify `project_id`.

Reply:
@blkt there is no project_id in this table. It points to a data_sources row which already has a project ID check

Reply:
Should we add it?